core/bones/record.py

```diff
@@ -60,7 +60,9 @@ def parseSubfieldsFromClient(self) -> bool:
         """
         return True
 
-    def singleValueFromClient(self, value, skel, name, origData):
+    def singleValueFromClient(self, value: Any, skel: 'SkeletonInstance',
+                              bone_name: str, client_data: dict
+                              ) -> tuple[Any, list[ReadFromClientError] | None]:
```

Comment:
Is it necessary to provide full type annotations in sub-classes as well?
I would like to keep maintenance overhead low, if possible.